File: docs/template_section_registry.py
from enum import Enum
from pydantic import BaseModel

class SectionSemanticFamily(str, Enum):
    IDENTITY = 'identity_section'
    DISCLOSURE = 'disclosure_section'
    PRICING = 'pricing_section'
    TRACEABILITY = 'traceability_section'
    REFLECTION = 'reflection_section'
    EVIDENCE = 'evidence_section'
    CONTINUITY = 'continuity_section'
    INTERPRETATION = 'interpretation_section'
    SIGNATURE = 'signature_section'

class SectionMutability(str, Enum):
    MUTABLE = 'mutable'
    READONLY = 'readonly'
    IMMUTABLE = 'immutable'
    SYSTEM_GENERATED = 'system_generated'
    STANDARD_MANDATED = 'standard_mandated'

class RegistryVersion(BaseModel):
    registry_version: str
    last_updated: str
    governance_revision: str

TEMPLATE_SECTION_REGISTRY_VERSION = RegistryVersion(
    registry_version="1.0.0",
    last_updated="2026-05-27",
    governance_revision="REV-001"
)

class TemplateSection(BaseModel):
    section_key: str
    section_label: str
    semantic_family: SectionSemanticFamily
    mutability: SectionMutability
    description: str

TEMPLATE_SECTION_REGISTRY = [
    TemplateSection(
        section_key='SEC-ID-001',
        section_label='Steward Context',
        semantic_family=SectionSemanticFamily.IDENTITY,
        mutability=SectionMutability.READONLY,
        description='Governed identity derived directly from the business profile.'
    ),
    TemplateSection(
        section_key='SEC-DISC-001',
        section_label='Standard Disclosure Notice',
        semantic_family=SectionSemanticFamily.DISCLOSURE,
        mutability=SectionMutability.IMMUTABLE,
        description='Mandatory footer injected by operational standards protecting against false authority.'
    ),
    TemplateSection(
        section_key='SEC-PRICE-001',
        section_label='Proposed Cost Breakdown',
        semantic_family=SectionSemanticFamily.PRICING,
        mutability=SectionMutability.MUTABLE,
        description='Detailed proposal of value exchange, editable by the steward.'
    ),
    TemplateSection(
        section_key='SEC-TRACE-001',
        section_label='Ingredient / Materials Sourcing',
        semantic_family=SectionSemanticFamily.TRACEABILITY,
        mutability=SectionMutability.STANDARD_MANDATED,
        description='Tracks origin of inputs. Required for food or industrial continuity templates.'
    ),
    TemplateSection(
        section_key='SEC-REFL-001',
        section_label='Steward Reflection',
        semantic_family=SectionSemanticFamily.REFLECTION,
        mutability=SectionMutability.MUTABLE,
        description='Captures teaching, pastoral notes, or operational wisdom.'
    ),
    TemplateSection(
        section_key='SEC-EVID-001',
        section_label='Visual Evidence Matrix',
        semantic_family=SectionSemanticFamily.EVIDENCE,
        mutability=SectionMutability.MUTABLE,
        description='Provides before/after or physical proof attached to the document.'
    ),
    TemplateSection(
        section_key='SEC-CONT-001',
        section_label='Replay Lineage Block',
        semantic_family=SectionSemanticFamily.CONTINUITY,
        mutability=SectionMutability.SYSTEM_GENERATED,
        description='Automatically renders the parent_event_id and timeline hashes anchoring the document to reality.'
    ),
]

File: archive_apps/mobile/data/templateSectionRegistry.ts
// Governed dictionary of semantic document building blocks.
// Prevents section sprawl and enforces strict mutability constraints for compliance.

export type SectionSemanticFamily =
    | 'identity_section'
    | 'disclosure_section'
    | 'pricing_section'
    | 'traceability_section'
    | 'reflection_section'
    | 'evidence_section'
    | 'continuity_section'
    | 'interpretation_section'
    | 'signature_section';

export type SectionMutability =
    | 'mutable'           // Steward can freely edit or remove in draft
    | 'readonly'          // Visible to steward, cannot be edited, but could be excluded if template allows
    | 'immutable'         // Cannot be edited, removed, or suppressed by steward or template
    | 'system_generated'  // Filled entirely by backend logic (e.g., Replay IDs)
    | 'standard_mandated'; // Forced into inclusion by an active standard, steward must fulfill it

export interface TemplateSection {
    sectionKey: string;
    sectionLabel: string;
    semanticFamily: SectionSemanticFamily;
    mutability: SectionMutability;
    description: string;
}

export const TEMPLATE_SECTION_REGISTRY: TemplateSection[] = [
    {
        sectionKey: 'SEC-ID-001',
        sectionLabel: 'Steward Context',
        semanticFamily: 'identity_section',
        mutability: 'readonly',
        description: 'Governed identity derived directly from the business profile.'
    },
    {
        sectionKey: 'SEC-DISC-001',
        sectionLabel: 'Standard Disclosure Notice',
        semanticFamily: 'disclosure_section',
        mutability: 'immutable',
        description: 'Mandatory footer injected by operational standards protecting against false authority.'
    },
    {
        sectionKey: 'SEC-PRICE-001',
        sectionLabel: 'Proposed Cost Breakdown',
        semanticFamily: 'pricing_section',
        mutability: 'mutable',
        description: 'Detailed proposal of value exchange, editable by the steward.'
    },
    {
        sectionKey: 'SEC-TRACE-001',
        sectionLabel: 'Ingredient / Materials Sourcing',
        semanticFamily: 'traceability_section',
        mutability: 'standard_mandated',
        description: 'Tracks origin of inputs. Required for food or industrial continuity templates.'
    },
    {
        sectionKey: 'SEC-REFL-001',
        sectionLabel: 'Steward Reflection',
        semanticFamily: 'reflection_section',
        mutability: 'mutable',
        description: 'Captures teaching, pastoral notes, or operational wisdom.'
    },
    {
        sectionKey: 'SEC-EVID-001',
        sectionLabel: 'Visual Evidence Matrix',
        semanticFamily: 'evidence_section',
        mutability: 'mutable',
        description: 'Provides before/after or physical proof attached to the document.'
    },
    {
        sectionKey: 'SEC-CONT-001',
        sectionLabel: 'Replay Lineage Block',
        semanticFamily: 'continuity_section',
        mutability: 'system_generated',
        description: 'Automatically renders the parent_event_id and timeline hashes anchoring the document to reality.'
    },
    {
        sectionKey: 'SEC-INT-001',
        sectionLabel: 'Bounded Interpretation',
        semanticFamily: 'interpretation_section',
        mutability: 'mutable',
        description: 'The steward’s meaning applied to facts (e.g., Claim Guidance, Sampling Outcome).'
    },
    {
        sectionKey: 'SEC-SIG-001',
        sectionLabel: 'Steward Commitment Sign-off',
        semanticFamily: 'signature_section',
        mutability: 'mutable',
        description: 'Human intent to proceed, anchoring the document to a specific steward session.'
    }
];
